src/core/namespace.c

```diff
@@ -909,6 +909,18 @@ static int clone_device_node(
         return 0;
 }
 
+static char *settle_runtime_dir(void) {
+        char *runtime_dir;
+
+        if (geteuid() == 0)
+                return strdup("/run/");
```

Comment:
this check is not right.

systemd --user instances run as root and should *not* use /run/ as runtime dir

Reply:
They don't run as root?

```
[Service]
User=%i
PAMName=systemd-user
Type=notify
ExecStart=/lib/systemd/systemd --user
```

Reply:
user@0.service means User=0 is set, so they very much run as root

Reply:
ah for the root instance, sure - it's a pre-existing condition though, the check was already like this for InaccessiblePaths

Reply:
Meh it's easy enough to fix will update

Reply:
> ah for the root instance, sure - it's a pre-existing condition though, the check was already like this for InaccessiblePaths

if you touch it you fix it ;-)

Reply:
actually, nope. the old code was OK. the geteuid() check was done to determine the mount source for the "inaccessible inodes" mounts. And for that it's entirely ok (and in fact should be left as is).

it's a different usecase basically: storing stuff for the --user instance outside of XDG_RUNTIME_DIR in /run is a bad idea. But consuming generic pre-prepared inodes from there, is OK.

Reply:
But I'd still like to make things consistent :)